core/src/impl/Kokkos_ClockTic.hpp

```diff
@@ -74,7 +74,10 @@ uint64_t clock_tic(void) noexcept {
 #elif defined(__HCC_ACCELERATOR__)
   // Get clock register
   return hc::__clock_u64();
-
+#elif defined(KOKKOS_ENABLE_OPENMPTARGET)
+  return (uint64_t)std::chrono::high_resolution_clock::now()
+      .time_since_epoch()
+      .count();
```

Comment:
Why is this block needed?

Reply:
because the CPU architecture flags which trigger the inclusion of assembly are defined during the device side compilation of OpenMPTarget. Thus we would try to insert invalid assembly, and need to intercept this with some standard code at the top.